agent/connect/generate_test.go

```diff
@@ -138,10 +139,8 @@ func TestSignatureMismatches(t *testing.T) {
 				continue
 			}
 			t.Run(fmt.Sprintf("TestMismatches-%s%d-%s%d", p1.keyType, p1.keyBits, p2.keyType, p2.keyBits), func(t *testing.T) {
-				ca := TestCAWithKeyType(t, nil, p1.keyType, p1.keyBits)
-				r.Equal(p1.keyType, ca.PrivateKeyType)
-				r.Equal(p1.keyBits, ca.PrivateKeyBits)
```

Comment:
These two assertions did nothing but test the test helper. We no longer return the key type and key bits, since those are passed in.